dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/Application/MeetingsTests.cs

```diff
@@ -0,0 +1,158 @@
+        {
+            // Arrange
+            var adapter = new NotImplementedAdapter();
+            var turnContexts = CreateMeetingTurnContext("application/vnd.microsoft.meetingParticipantJoin", adapter);
+            var app = new Application<TestTurnState, TestTurnStateManager>(new()
+            {
+                RemoveRecipientMention = false,
+                StartTypingTimer = false
+            });
+            var ids = new List<string>();
+            app.Meetings.OnParticipantsJoin((context, _, _, _) =>
+            {
+                ids.Add(context.Activity.Id);
+                return Task.CompletedTask;
+            });
+
+            // Act
+            foreach (var turnContext in turnContexts)
+            {
+                await app.OnTurnAsync(turnContext);
+            }
+
+            // Assert
+            Assert.Single(ids);
+            Assert.Equal("test.id", ids[0]);
+        }
+
+        [Fact]
+        public async void Test_OnParticipantsLeave()
+        {
+            // Arrange
+            var adapter = new NotImplementedAdapter();
+            var turnContexts = CreateMeetingTurnContext("application/vnd.microsoft.meetingParticipantLeave", adapter);
+            var app = new Application<TestTurnState, TestTurnStateManager>(new()
+            {
+                RemoveRecipientMention = false,
+                StartTypingTimer = false
+            });
+            var ids = new List<string>();
+            app.Meetings.OnParticipantsLeave((context, _, _, _) =>
+            {
+                ids.Add(context.Activity.Id);
+                return Task.CompletedTask;
+            });
+
+            // Act
+            foreach (var turnContext in turnContexts)
+            {
+                await app.OnTurnAsync(turnContext);
+            }
+
+            // Assert
+            Assert.Single(ids);
+            Assert.Equal("test.id", ids[0]);
+        }
+
+        private static TurnContext[] CreateMeetingTurnContext(string activityName, BotAdapter adapter)
+        {
+            return new TurnContext[]
+            {
+                new(adapter, new Activity
+                {
+                    Type = ActivityTypes.Event,
+                    ChannelId = Channels.Msteams,
+                    Name = activityName,
+                    Id = "test.id"
+                }),
+                new(adapter, new Activity
+                {
+                    Type = ActivityTypes.Event,
+                    ChannelId = Channels.Msteams,
+                    Name = "fake.name"
+                }),
+                new(adapter, new Activity
+                {
+                    Type = ActivityTypes.Invoke,
+                    ChannelId = Channels.Msteams,
+                    Name = activityName
+                }),
+                new(adapter, new Activity
+                {
+                    Type = ActivityTypes.Event,
+                    ChannelId = Channels.Webchat,
```

Comment:
Should this be `Webchat` or `Msteams`?

Reply:
This is a test case for non-teams activity, so use `Webchat` as test value.